src/threading/thread.cpp

```diff
@@ -71,7 +71,32 @@ Thread::Thread(const std::string &name) :
 
 Thread::~Thread()
 {
-	kill();
+	// kill the thread if running
+	if (!m_running) {
+		wait();
+	} else {
+
+		m_running = false;
```

Comment:
as you are in destructor, you can drop all asignment there because after destructor we won't read them :D